docs/custom/error-handlers.mdx

```diff
@@ -0,0 +1,154 @@
+AMF allows you to define custom error handlers that you can customize and plug into AMF's resolution stage (and in the future, all stages).
+
+There are 3 types of errors in AMF, denoted by the `amf.core.validation.SeverityLevels` object:
+
+- **Info**
+  - This severity level points to something that may or may not be wrong
+  - Usually depends on user intent or use case
+- **Warning**
+  - This severity level indicates to the user that something is not correct, and should be changed
+  - Usually warns about future changes, using deprecated tools, or errors that will become violations the future
+- **Violation**
+  - This severity level indicates that something is definitely wrong
+  - Indication that something is not compliant with its specification
+  - This severity makes the AMF graph model invalid
+
+# Default error handling process
+AMF internally transforms each error into an `ValidationResult`, then collects all results generated in a `ValidationReport`
+that is returned after the validation stage, as seen in [previous examples](../AMF/using-amf/amf_validation#validation-examples).
+
+```scala title="amf.core.validation.core.ValidationReport"
+trait ValidationResult {
+  def message: Option[String]
+  def path: String
+  def sourceConstraintComponent: String
+  def focusNode: String
+  def severity: String
+  def sourceShape: String
+}
+
+trait ValidationReport {
+  def conforms: Boolean
+  def results: List[ValidationResult]
+}
+```
+## ValidationReport example
+For example, consider the following OAS 3.0 API:
+
+```json
+{
+  "openapi": "3.0.0",
+  "info": {
+    "title": "test api",
+    "version": "1"
+  }
+}
+```
+
+This API is invalid as it doesn't have the `paths:{}` object that is required by the
+[OpenAPI 3.0 Specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#schema) to consider it a valid API,
+and validating it with AMF returns the following ValidationReport:
+
+```text
+Profile: OAS 3.0
+Conforms? false
+Number of results: 1
+- Source: http://a.ml/vocabularies/amf/parser#mandatory-paths-property
+  Message: 'paths' is mandatory in OAS spec
+  Level: Violation
+  Target: file://amf-client/shared/src/test/resources/validations/oas3/paths-property.json#/web-api
+  Property:
+  Position: None
+  Location:
+```
+
+
+This report contains detailed information about all errors found, but there are still several reasons why you'd want a custom error handler:
+
+- You want to stop execution when a certain kind of error is found
+- You want to process, modify or log incoming errors
+- You want to run and check errors of a single stage (e.g. parse an OAS 2.0 API and check if it has parsing errors)
+
+In case you want custom functionality, you need to implement the provided **error handler interface**.
+
+## Error handler interface
+The error handler interface is currently located at `amf.client.resolve`, as it is supported only in the resolution stage.
```

Comment:
The error handler interface is located at `amf.client.resolve`, as it's used only in the resolution stage.
The error handler interface declares the method **`reportConstraint()`**, which AMF uses to report runtime errors: